lib/motors/PID.cpp

```diff
@@ -0,0 +1,56 @@
+#include "PID.hpp"
+
+void PIDController::PIDController_Init()
+{
+  /* Clear controller variables */
+  integrator      = 0.0f;
+  prevError       = 0.0f;
+  differentiator  = 0.0f;
+  prevMeasurement = 0.0f;
+
+  out = 0.0f;
+};
+
+float PIDController::PIDController_Update(float setpoint, float measurement)
+{
+  // Error signal
+  float error = setpoint - measurement;
+
```

Comment:
lots of unnecessary whitespace throughout, generally avoid it. (if you have a big enough block of code for that to get unpleasant then you should refactor into functions)